### ISSUE TITLE
i am getting ADF is not defined error in  2.0

### ISSUE THREAD

**Issue by dhrn** [Alfresco/alfresco-ng2-components#2754]
i am using adf components 2.0 
I am using adf-form component with stencils. ie I am using stencils within the form.
i did the changes as per this documentation -> _[Replacing custom stencils with custom components](https://github.com/Alfresco/alfresco-ng2-components/blob/master/docs/extensibility.md)_
My code
```
          this.formRenderingService.setComponentTypeResolver('data_grid', () => GridComponent, true);
          this.formRenderingService.setComponentTypeResolver('detailgrid', () => DetailComponent, true);
          this.formRenderingService.setComponentTypeResolver('matching_grid', () => MathchingComponent, true);
          this.formRenderingService.setComponentTypeResolver('sample_grid', () => SampleComponent, true);
          this.formData = this.formService.parseForm(res1);
```
` <adf-form [form]="formData" (click)="onFieldClick($event)" (formSaved)="formSaved()" (formCompleted)="formCompleted()"></adf-form>`

i am getting the following error

> ContainerWidgetComponent.html:17 ERROR ReferenceError: adf is not defined
>     at FormFieldComponent.hasController (adf-core.umd.js:17579)
>     at FormFieldComponent.ngOnInit (adf-core.umd.js:17529)
>     at checkAndUpdateDirectiveInline (core.js:12094)
>     at checkAndUpdateNodeInline (core.js:13597)
>     at checkAndUpdateNode (core.js:13540)
>     at debugCheckAndUpdateNode (core.js:14412)
>     at debugCheckDirectivesFn (core.js:14353)
>     at Object.eval [as updateDirectives] (ContainerWidgetComponent.html:17)
>     at Object.debugUpdateDirectives [as updateDirectives] (core.js:14338)
>     at checkAndUpdateView (core.js:13507)

i am getting this error in 2.0 but it is running without any error's in 1.9.0

**Comment by eromano** [comment]
Hi @dhrn can you provide a sandbox app where we can test this problem?

**Comment by DenisVuyka** [comment]
You may get this error only if you try to provide AngularJS controllers with the stencils.

**Comment by dhrn** [comment]
i am confused now with these tutorials
https://github.com/Alfresco/alfresco-ng2-components/blob/master/docs/stencils.md
https://github.com/Alfresco/alfresco-ng2-components/blob/master/docs/extensibility.md
which documentation i have to follow?. please give me the correct documentation for this. some deprecated lines are still there in the stencils((https://github.com/Alfresco/alfresco-ng2-components/blob/master/docs/stencils.md)

**Comment by dhrn** [comment]
thanks @DenisVuyka  i have removed angular 1 controller in APS, now it is working in ADF. But now In APS it  is not working.

**Comment by dhrn** [comment]
hi @DenisVuyka , After removing AngularJS controller in APS, it is working fine. But in APS, it is not working.Is there any solution for this APS Issue. can we use jquery(or something) to achieve this. suggest a solution for this problem.

**Comment by DenisVuyka** [comment]
you can try adding the following script to the ADF app (index.html):

```
window.adf = { components: {} };
```

that should do the trick with controllers and you  can use regular stencils.
We are working on a more elegant solution.

**Comment by DenisVuyka** [comment]
fixed and will be available with 2.1 release
